worksheet.rb

```diff
@@ -31,4 +31,147 @@
+                cost: 45,
+                rider: "RD0003",
+                rating: 2
+            }
+        ],
+        dr0002: [
+            {
+                date: "3rd Feb 2016",
+                cost: 25,
+                rider: "RD0073",
+                rating: 5
+            },
+            {
+                date: "4th Feb 2016",
+                cost: 15,
+                rider: "RD0013",
+                rating: 1
+            },
+            {
+                date: "5th Feb 2016",
+                cost: 35,
+                rider: "RD0066",
+                rating: 3
+            }
+        ],
+        dr0003: [
+            {
+                date: "4th Feb 2016",
+                cost: 5,
+                rider: "RD0066",
+                rating: 5
+            },
+            {
+                date: "5th Feb 2016",
+                cost: 50,
+                rider: "RD0003",
+                rating: 2
+            }
+        ],
+        dr0004: [
+            {
+                date: "3rd Feb 2016",
+                cost: 5,
+                rider: "RD0022",
+                rating: 5
+            },
+            {
+                date: "4th Feb 2016",
+                cost: 10,
+                rider: "RD0022",
+                rating: 4
+            },
+            {
+                date: "5th Feb 2016",
+                cost: 20,
+                rider: "RD0073",
+                rating: 5
+            }
+        ]
+    }
+
+#Iterate through the data structure to display, by printing in terminal, the following info
+# The number of rides each driver has given
+# count each array
+puts "Number of rides each driver has given:"
+rides.each do |driver,trips|
+  puts "#{driver} has given #{trips.length} rides."
+end
+
+puts "______________________________________________________________"
+puts "**************************************************************"
+
+#The total amount of money each driver has made
+# for each driver loop over the array and give me the cost and add it to sum
+# which driver made the most money
+# optional: for each driver, on which day did they make the most money?
+puts "The total amount of money each driver has made:"
+max = 0
+max_driver = ""
+#date_max_money = ""
+ rides.each do |driver,trips|
+   sum = 0
+   puts "#{driver} earnings:"
+   trips.each do | trip |
+     puts "$#{trip[:cost].to_f}"
+     sum +=  trip[:cost].to_f
+     if sum > max
+       max = sum
+       max_driver = driver
+     end
+   end
+   puts "Total earnings: $#{sum}."
+ end
+puts "#{max_driver} made the most money in the amount of $#{max} dollars."
+
+puts "______________________________________________________________"
+puts "**************************************************************"
+
+
+#The average rating for each driver
+# average = sum of terms / number of terms
+#Which driver has the highest average rating?
+puts "The average rating for each driver:"
+highest_average = 0
+driver_highest_avg = ""
```

Comment:
This is just a suggestion, or something to think of:

Here, I see you are initializing `driver_highest_avg` with an empty string `""`. This is because you want kind of a "blank/empty default value"! This is the right thing to go for.

As a future suggestion, you might consider using `nil`... or even `0` in this case, since an average's default value should be a number!

The best "default value" depends on the context, and an empty string isn't wrong, but it's something we can think about in the future!